### PR TITLE
add draft network

### PR DESCRIPTION
**Pull request by iamai74**

### CODE REVIEW

File: BrainUp/BrainUp/Network/NetworkService.swift [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+//  NetworkService.swift
+//  BrainUp
+//
+//  Created by Evgenii Zhigunov on 2/18/22.
+//
+
+import Foundation
+
+protocol NetworkService {
+    func fetch<T: Codable>(_ request: Request, completion: @escaping (Result<T, Error>) -> Void)
```
Comment (Venteor) [inline BrainUp/BrainUp/Network/NetworkService.swift]: Should we use here Codable? I suggest use only Decodable.

File: BrainUp/BrainUp/Network/AlamofireNetworkService.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+//  BrainUp
+//
+//  Created by Evgenii Zhigunov on 2/18/22.
+//
+
+import Foundation
+import Alamofire
+
+class AlamofireNetworkService: NetworkService {
+    func fetch<T>(_ request: Request, completion: @escaping (Result<T, Error>) -> Void) where T: Decodable, T: Encodable {
```
Comment (Venteor) [inline BrainUp/BrainUp/Network/AlamofireNetworkService.swift]: Should we use here Encodable? I suggest use only Decodable.

**Review by Venteor — COMMENTED** [review]
Everything is good. Just some minor question regarding Codable and Decodable.

File: BrainUp/BrainUp/Network/NetworkService.swift (hunk 1 above)
Comment (iamai74) [reply BrainUp/BrainUp/Network/NetworkService.swift]: fixed

File: BrainUp/BrainUp/Network/AlamofireNetworkService.swift (hunk 2 above)
Comment (iamai74) [reply BrainUp/BrainUp/Network/AlamofireNetworkService.swift]: fixed

**Review by Venteor — APPROVED** [review]

**Pull request merged**